woocommerce/woocommerce-ios
# [Blaze] Payment methods list screen

Closes: #11717 


## Description

Adds the screen that lists the saved payment methods from where the user can select a payment method for the Blaze campaign creation flow. Also, the user can add a new payment method from this screen.

Apologies for the 500+ lines of changes. They are due to unit tests. 

#### Changes
- Add a new view and view model for listing Blaze payment methods
- Change the `BlazeConfirmPaymentView` and its view model to allow payment selection. 
- Unit tests

#### Note

The adding payment method feature hasn't been tested yet. I added a separate subtask to test adding a payment method. We can pick up this work this once the API is ready. 

## Testing instructions
- Log in to a store that's eligible for Blaze.
- Navigate to the Products tab, select a published product and tap Promote with Blaze.
- On the campaign creation form, tap Confirm Details.
- Tap on the card info row to open the payment method list screen
- Verify that the WPCOM account name, email are displayed correctly in the list footer.
- Change the payment method, tap `Save`, and verify that the selection is saved.

## Screenshots

#### No payment methods available
| Dark | Light |
|--------|--------|
| <img width="320" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/ce3f05a2-6776-4d95-9545-72f885816003"> | <img width="323" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/575dbf97-076f-41a6-a51d-89a83dcf40b0"> | 

#### Payment methods available
| Dark | Light |
|--------|--------|
| <img width="324" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/4b2b9030-4b9c-436e-951d-95ce64f90e2c"> | <img width="327" alt="image" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/072b1580-2abc-404d-8bd2-34d5ee16d56d"> | 

![Simulator Screen Recording - iPhone 15 Pro - 2024-01-18 at 20 37 53](https://github.com/woocommerce/woocommerce-ios/assets/524475/6b101821-2642-478f-9b32-50fca50a5a65)

---
